mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp

```diff
@@ -85,25 +86,20 @@ LogicalResult PopulateParams::calculateBlockGemmPerformanceParameters(
         param.gemmNPerBlock % param.gemmNPerThread == 0))
     return failure();
 
-  int64_t threadGemmMPerBlock = param.gemmMPerBlock / param.gemmMPerThread;
-  int64_t threadGemmNPerBlock = param.gemmNPerBlock / param.gemmNPerThread;
-
-  int64_t threadGemmMPerCluster =
-      derived.mThreadsPerCuwave * derived.mCuwavesPerBlock;
-  int64_t threadGemmNPerCluster =
-      derived.nThreadsPerCuwave * derived.nCuwavesPerBlock;
-
-  if (!(threadGemmMPerBlock % threadGemmMPerCluster == 0) &&
-      (threadGemmNPerBlock % threadGemmNPerCluster == 0))
-    return failure();
-
-  int64_t clusterMPerBlock = threadGemmMPerBlock / threadGemmMPerCluster;
-  int64_t clusterNPerBlock = threadGemmNPerBlock / threadGemmNPerCluster;
-
-  // inline asm only support clusterMPerBlock = 2 andclusterNPerBlock =
-  // 2
-  if (!(clusterMPerBlock == 2 && clusterNPerBlock == 2))
+  int64_t threadGemmMPerCluster = param.gemmMPerThread *
+                                  derived.mThreadsPerCuwave *
+                                  derived.mCuwavesPerBlock;
+  int64_t threadGemmNPerCluster = param.gemmNPerThread *
+                                  derived.nThreadsPerCuwave *
+                                  derived.nCuwavesPerBlock;
+
+  if ((param.gemmMPerBlock % threadGemmMPerCluster != 0) ||
+      (param.gemmNPerBlock % threadGemmNPerCluster != 0)) {
+    LLVM_DEBUG(
+        llvm::dbgs()
+        << "M per block or N per block aren't divisible by M/N per cluster\n");
     return failure();
```

Comment:
I have no idea why there was the magic number of 2 and 2. It does seem to be a workaround from CK that we just bring it over regardless.

Reply:
Yeah, idk either. Especially since the comment said something about inline ASM